htdocs/societe/card.php

```diff
@@ -404,7 +404,7 @@
 	        }
 	        $object->entity					= (GETPOSTISSET('entity') ? GETPOST('entity', 'int') : $conf->entity);
 	        $object->name_alias = GETPOST('name_alias', 'alphanohtml');
-	        $object->address				= GETPOST('address', 'alphanohtml');
+	        $object->address				= GETPOST('address', 'none');
```

Comment:
I think the trouble is not in param but in implementation when param is alphanohtml.
Can you try if this fix done on v12 solve your trouble : c48a294a2ba25a326ec8535a0d19ae5e59bdfe84
If yes a cherry-pick would be better that allowing any data without sanitization.

Reply:
I checked this out. This is not related to the commit you mention, but to this one : 8b879ee23b58860733abc76a41f5ee8f66ef936f
The dol_string_nohtmltag was called with parameter that suppressed line break. So all textarea fields with alphanohtml param were losing the \r\n. So either we replicate this commit in V11, or we use simple "alpha" param for textarea fields

Reply:
You're right, i pasted the wrong commit id. Good one is 8b879ee
To know if with use alpha or alphanohtml, we must think what is the goal of field. We must use in priority "alphanohtml" and use alpha ONLY if allowing HTML inside text is necessary.
So we must backport the $out = dol_string_nohtmltag($out, 0); instead of $out = dol_string_nohtmltag($out, 1);
instead of changing the param of sanitizing function. Because param of sanitizing function depends on context and type of data we accept, so there is no reason to change it (except if it's wrong, but for an address field, alphanohtml seems good to me).